### PR TITLE
Update dependencies and python constraints

### PR DESCRIPTION
**Pull request by xylar**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by xylar** [comment]
I am having trouble with `pip check` on an earlier version of this package.  I want to fix things up for the current version first, and then make similar changes for v3.10.10 used by apache beam.  See:
https://github.com/conda-forge/apache-airflow-providers-apache-beam-feedstock/pull/35

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -126,3 +128,4 @@ extra:
     - sodre
     - xhochy
     - ocefpaf
+    - xylar
```
Comment (xylar) [inline recipe/meta.yaml:131]: If you think another maintainer would be helpful, I'm happy to contribute.  If you'd prefer not to have me, no problem.

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 2]
```diff
@@ -32,25 +32,27 @@ requirements:
...
   run:
     - propcache >=0.2.0
     - aiohappyeyeballs >=2.3.0
-    - python         # [not use_noarch]
-    - python >=3.8   # [use_noarch]
+    - python                     # [not use_noarch]
+    - python >={{ python_min }}  # [use_noarch]
     - aiosignal >=1.1.2
-    - async-timeout >=4.0,<5.0  # [py<311]
+    - async-timeout >=4.0,<6.0   # [py<311 or use_noarch]
```
[the remark this answers is not in the document]
Comment (xhochy) [reply recipe/meta.yaml:46]: If this doesn't auto-update, should we disable `automerge`?

**Review by xhochy — APPROVED** [review]

File: recipe/meta.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (xylar) [reply recipe/meta.yaml:46]: Yes, I think we should.  I'll add that here.

**Comment by xylar** [comment]
Done!

**Comment by xylar** [comment]
In general, I think grayskull isn't doing well with `pyproject.toml` files that have different dependencies for different python versions.  I've looked into fixing this but it seems like a can of worms.  @xhochy, I think maybe you've looked into it as well in the past and realized the same?  There certainly was someone from Core who had.

**Comment by xylar** [comment]
@ocefpaf, @xhochy, I think this is now ready to merge.  Thanks for having a look!

**Comment by xhochy** [comment]
> In general, I think grayskull isn't doing well with `pyproject.toml` files that have different dependencies for different python versions. I've looked into fixing this but it seems like a can of worms. @xhochy, I think maybe you've looked into it as well in the past and realized the same? There certainly was someone from Core who had.

I looked a bit into it but wasn't able to figure out a solution in reasonable time (for me).

**Pull request merged**